test/test_acceptance.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+from selenium import webdriver
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions
+import time
+import pytest
+
+SECONDS_WAIT = 15
+
+
+@pytest.fixture()
+def driver(pytestconfig):
+    browser = pytestconfig.getoption("browser")
+    headless = pytestconfig.getoption('headless')
+
+    if browser == 'firefox':
+        options = webdriver.FirefoxOptions()
+        if headless:
+            options.headless = True
+        driver = webdriver.Firefox(options=options)
+    elif browser == 'chrome':
+        options = webdriver.ChromeOptions()
+        if headless:
+            options.headless = True
+        driver = webdriver.Chrome(options=options)
+
+    else:
+        raise Exception(f'No browser option available for {browser}')
+
+    yield driver
+
+    driver.quit()
+
+
+def test_authentication(driver):
+    driver.get('http://localhost:8000/hub/saml_login')
+    WebDriverWait(driver, SECONDS_WAIT).until(
```

Comment:
Could use a helper function to remove repeated wait code.

Reply:
Sounds good